src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -9193,5 +9193,26 @@ void M(S s)
                 //         int* f = &(s_f.Buf[0]);
                 Diagnostic(ErrorCode.ERR_FixedNeeded, "&(s_f.Buf[0])").WithLocation(17, 18));
         }
+
+        [Fact, WorkItem(34693, "https://github.com/dotnet/roslyn/issues/34693")]
+        public void Repro_34693()
+        {
+            var csharp = @"
+namespace Interop
+{
+    public unsafe struct PROPVARIANT
+    {
+        public CAPROPVARIANT ca;
+    }
+
+    public unsafe struct CAPROPVARIANT
+    {
+        public uint cElems;
+        public PROPVARIANT* pElems;
+    }
+}";
+            var comp = CreateCompilation(csharp, options: TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics();
```

Comment:
What about the other test case about accessing a field through a derived type? Is that an existing test or do we need to add it?

Reply:
I can add the coverage however it's basically going to just reveal a less-impactful bug that was introduced in `features/unmanaged-constructed-types` as a side effect of delaying constraint checks until after field binding.

Reply:
There's one that's specific to "error on pointer to managed type" for which coverage already exists. We didn't happen to test a similar scenario for fields of a constrained generic type.

B<string*[]>.T, where T is not declared in B, is a Heisentype.